docs/contributing/add_new_resource_tgc_next.md

```diff
@@ -0,0 +1,330 @@
+	if asset == nil || asset.Resource == nil || asset.Resource.Data == nil {
+		return nil, fmt.Errorf("asset resource data is nil")
+	}
+
+	hclData := make(map[string]interface{})
+	// copy values from the CAI asset to hclData
+	// return any errors encountered
+	// ...
+
+	ctyVal, err := common.MapToCtyValWithSchema(hclData, c.schema)
+	if err != nil {
+		return nil, err
+	}
+	return &common.HCLResourceBlock{
+		Labels: []string{c.name, "Whatever"},
+		Value:  ctyVal,
+	}, nil
+}
+
+```
+
+You will also need to add an entry to [`tgc_next/cai2hcl/resource_converters.go.tmpl`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/tgc_next/cai2hcl/resource_converters.go.tmpl), which is used to generate [`pkg/cai2hcl/converters/resource_converters.go`](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/main/pkg/cai2hcl/converters/resource_converters.go). Each entry in `resource_converters.go.tmpl` maps a CAI asset type to a function that returns a ResourceConverter - in this case:
+
+```golang
+// ...
+"alloydb.googleapis.com/Backup": {
+	"Default": alloydb.NewAlloydbBackupCai2hclConverter(provider),
+},
+// ...
+```
+
+#### 2. Generate terraform-google-conversion
+To generate terraform-google-conversion code locally, run the following from the root of the `magic-modules` repository:
+
+```
+make tgc OUTPUT_PATH="/path/to/your/terraform-google-conversion"
+```
+
+#### 3. Run tests locally
+##### Before you begin
+1. Set the following environment variable:
+```
+export WRITE_FILES=true
+```
+
+2. Please request the necessary permission for the GCS bucket [cai_assets_metadata](https://pantheon.corp.google.com/storage/browser/cai_assets_metadata;tab=permissions?forceOnBucketsSortingFiltering=true&e=13802955&inv=1&invt=Ab0TPA&mods=ml_engine_alpha_ng2&project=tgc-test-infra&prefix=&forceOnObjectsSortingFiltering=false), which stores the nightly test metadata required to run the integration tests.
+
+	It might require to install relevant go version for this to run successfully. Follow the [Before you begin](https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/#before-you-begin) and [Setup your development environment](https://googlecloudplatform.github.io/magic-modules/develop/set-up-dev-environment/) sections from the [Magic Modules documentation](https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/).
+
+##### Run unit tests
+To run the unit tests locally, run the following from the root of the `terraform-google-conversion` repository:
+
+```
+make test
+```
+
+##### Run integration tests
+To run the integration tests for the added resource locally, run the following from the root of the `terraform-google-conversion` repository:
+```
+make test-integration TESTPATH=./test/services/alloydb  TESTARGS='-run=TestAccAlloydbBackup_alloydbBackupBasicTestExample' > alloydbBackup.log
```

Comment:
This example is for running a single test, we probably want to note that running all the tests for the resource can be done with `TESTARGS='-run=TestAccAlloydbBackup`